src/VirtualClient/VirtualClient.Actions/HadoopTerasort/HadoopTerasortExecutor.cs

```diff
@@ -0,0 +1,381 @@
+            DependencyPath javaPackage = await this.GetPackageAsync(
+                this.JdkPackageName, cancellationToken);
+
+            this.PackageDirectory = workloadPackage.Path;
+            this.JavaPackageDirectory = javaPackage.Path;
+
+            this.JavaExecutablePath = this.PlatformSpecifics.Combine(this.JavaPackageDirectory, "bin", "java");
+
+            switch (this.Platform)
+            {
+                case PlatformID.Unix:
+                    this.ExecutablePath = this.PlatformSpecifics.Combine(this.PackageDirectory, "bin", "hadoop");
+                    break;
+
+                default:
+                    throw new WorkloadException(
+                        $"The Hadoop workload is not supported on the current platform/architecture " +
+                        $"{PlatformSpecifics.GetPlatformArchitectureName(this.Platform, this.CpuArchitecture)}.",
+                        ErrorReason.PlatformNotSupported);
+            }
+
+            if (!this.state.HadoopExecutorStateInitialized)
+            {
+                this.SetEnvironmentVariable(EnvironmentVariable.JAVA_HOME, this.JavaPackageDirectory, EnvironmentVariableTarget.Process);
+
+                this.ConfigurationFilesAsync(telemetryContext, cancellationToken);
+                await this.ExecutableFilesAsync(telemetryContext, cancellationToken);
+
+                await this.ExecuteCommandAsync("bash", $"-c \"echo y | ssh-keygen -t rsa -P '' -f ~/.ssh/id_rsa\"", this.PackageDirectory, telemetryContext, cancellationToken);
+                await this.ExecuteCommandAsync("bash", $"-c \"cat ~/.ssh/id_rsa.pub >> ~/.ssh/authorized_keys\"", this.PackageDirectory, telemetryContext, cancellationToken);
+                await this.ExecuteCommandAsync("bash", $"-c \"chmod 0600 ~/.ssh/authorized_keys\"", this.PackageDirectory, telemetryContext, cancellationToken);
+                await this.ExecuteCommandAsync("bash", $"-c \"echo y | bin/hdfs namenode -format\"", this.PackageDirectory, telemetryContext, cancellationToken);
+                await this.ExecuteCommandAsync("bash", $"-c \"bin/hdfs dfs -mkdir /user\"", this.PackageDirectory, telemetryContext, cancellationToken);
+                await this.ExecuteCommandAsync("bash", $"-c \"bin/hdfs dfs -mkdir /user/azureuser\"", this.PackageDirectory, telemetryContext, cancellationToken);
+                this.state.HadoopExecutorStateInitialized = true;
+            }
+
+            if (!this.state.HadoopExecutorServicesStarted)
+            {
+                await this.ExecuteCommandAsync("bash", $"-c sbin/start-dfs.sh", this.PackageDirectory, telemetryContext, cancellationToken);
+                await this.ExecuteCommandAsync("bash", $"-c sbin/start-yarn.sh", this.PackageDirectory, telemetryContext, cancellationToken);
+                this.state.HadoopExecutorServicesStarted = true;
+                await Task.Delay(30000); // Delay during the safe mode of the name node to verify the status of replicated nodes.
```

Comment:
Where do we got this duration from? do we need to adjust this duration for smaller or larger machines? I would be more ideal to have a deterministic way of doing this

Reply:
It is for operations that includes executing tasks rapidly, where encompassing swift initiation and termination, is crucial. A delay would help avoid potential failures in the process of loading data into the service. I have removed the task delay and could be considered later if errors are noticed later.